tools/kvm_deploy.sh

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env bash
+#
+# SPDX-License-Identifier: MIT
+#
+# Copyright (c) 2020 The Authors.
+# Authors: Bin Liang  <@liangbin>
+#
+# Summary: script to deploy network and KVM vms
+#
+
+#set -x
+
+# ask if script runs without sudo or root priveleges
+if [ "$(whoami)" != "root" ]; then
+    echo " you need to run this script as root"
+    exit 1
+fi
+
+if [ "$#" -eq 0 ]; then
+    echo " Please add parameter to tell me how many VMs to deploy"
+    exit 1
+fi
+
+CNT=$1
+
+echo "========= Step 1: check dependencies ..."
+MISSING=""
+FOUND=""
+checkdep() {
+    local exe="$1" package="$2" upstream="$3"
+    if command -v "$1" >/dev/null 2>&1; then
+        FOUND="${FOUND:+${FOUND} }$exe"
+        return "0"
+    fi
+    MISSING=${MISSING:+${MISSING }$package}
+    echo "missing $exe."
+    echo "  It can be installed in package: $package"
+    [ -n "$upstream" ] &&
+        echo "  Upstream project url: $upstream"
+    return 1
+}
+
+checkdep cloud-localds cloud-image-utils
+checkdep virt-install virtinst
+checkdep qemu-img qemu-kvm
+checkdep virsh libvirt-bin
+checkdep wget wget
+
+if [ -n "$MISSING" ]; then
+    echo
+    [ -n "${FOUND}" ] && echo "found: ${FOUND}"
+    echo "install missing deps with:"
+    echo "  apt-get update && apt-get install ${MISSING}"
```

Comment:
Do we want to add -y here to the apt-gets to skip user confirmation?

Reply:
agree